### PR TITLE
Update validator storage to better support large miner indexes.

### PR DESCRIPTION
**Pull request by Sid-Data-Universe**

### CODE REVIEW

File: storage/validator/mysql_validator_storage.py [hunk 1]
```diff
@@ -120,6 +120,22 @@ def _get_or_insert_label(self, label: str) -> int:
 
         return label_id
 
+    def _get_label_value_to_id_dict(self) -> Dict[str, int]:
+        """Gets an encountered label or stores a new one returning this Validator's unique id for it"""
```
Comment (RusticLuftig) [inline storage/validator/mysql_validator_storage.py:124]: nit: update

File: storage/validator/mysql_validator_storage.py [hunk 2]
```diff
@@ -143,7 +161,14 @@ def upsert_miner_index(self, index: MinerIndex):
...
 
             # Get or insert this Validator's labelId for the specified label.
-            label_id = self._get_or_insert_label(label)
+            label_id = -1
+            if label in label_value_to_id_dict:
+                # Use the already known value if available.
+                label_id = label_value_to_id_dict[label]
+            else:
+                # Else add to the label table and use that id.
+                label_id = self._get_or_insert_label(label)
```
Comment (RusticLuftig) [inline storage/validator/mysql_validator_storage.py:170]: Is it worth performing a batch insert and get? Not sure how likely this is, but if I miner shows up with 10k new labels that we haven't yet seen, we'll hit the perf issue again

File: storage/validator/mysql_validator_storage.py (hunk 1 above)
Comment (Sid-Data-Universe) [reply storage/validator/mysql_validator_storage.py:124]: Done.

File: storage/validator/mysql_validator_storage.py (hunk 2 above)
Comment (Sid-Data-Universe) [reply storage/validator/mysql_validator_storage.py:170]: Done.

**Review by RusticLuftig — APPROVED** [review]

**Pull request merged**